bounded/src/set.rs

```diff
@@ -440,35 +434,138 @@ fn translate<'a>(
                 .unwrap()
                 .mutable
             {
-                return Err(TranslationError::UpdateViolatedImmutability(
-                    indices_to_sets[update.index].to_string(),
-                ));
+                panic!("one of the generated updates violated immutability");
             }
         }
-        // check that all of the transitions can be run
+        // check that all guards and updates target different indices
         for a in &tr.guards {
-            if tr.guards.iter().any(|b| {
-                matches!((a, b), (
-                    Guard { index: a, value: false },
-                    Guard { index: b, value: true },
-                )
-                | (
-                    Guard { index: a, value: true, },
-                    Guard { index: b, value: false, },
-                ) if a == b)
-            }) {
-                panic!("found an untakeable transition\n{tr:?}");
+            if tr.guards.iter().any(|b| a != b && a.index == b.index) {
+                panic!("found two guards with the same index\n{tr:?}");
             }
         }
-        // check that all redundant updates have been removed
+        for a in &tr.updates {
+            if tr.updates.iter().any(|b| a != b && a.index == b.index) {
+                panic!("found two updates with the same index\n{tr:?}");
+            }
+        }
+        // check no updates are redundant
         for &Guard { index, value } in &tr.guards {
-            if tr.updates.contains(&Update { index, value }) {
-                panic!("found an unremoved redundant update")
+            if tr.updates.contains(&Update {
+                index,
+                formula: match value {
+                    true => Formula::always_true(),
+                    false => Formula::always_false(),
+                },
+            }) {
+                panic!("found an update that was redundant with a guard")
             }
         }
+        for Update { index, formula } in &tr.updates {
+            if let Formula::Guard(guard) = formula {
+                if *index == guard.index && guard.value {
+                    panic!("found an update that doesn't do anything")
+                }
+            }
+        }
+        assert!(tr.slow_guard != Formula::always_false());
```

Comment:
Since all the similar panics above have a nice-ish message, can you put a message on this assert?